FinancialTransactionsApi/V1/Helpers/FileGenerator.cs

```diff
@@ -0,0 +1,131 @@
+            var globalSettings = new GlobalSettings
+            {
+                ColorMode = ColorMode.Color,
+                Orientation = Orientation.Portrait,
+                PaperSize = PaperKind.A4,
+                Margins = new MarginSettings { Top = 10 },
+                DocumentTitle = $"{name} Statement Report"
+            };
+            var objectSettings = new ObjectSettings
+            {
+                PagesCount = true,
+                HtmlContent = TemplateGenerator.GetHTMLReportString(report),
+                WebSettings = { DefaultEncoding = "utf-8", UserStyleSheet = Path.Combine(Directory.GetCurrentDirectory(), "assets", "styles.css"), LoadImages = true },
+                HeaderSettings = { FontName = "Arial", FontSize = 9, Right = "Page [page] of [toPage]", Line = true },
+                FooterSettings = { FontName = "Arial", FontSize = 9, Line = true, Center = $"{name} Statement Report" }
+            };
+            var pdf = new HtmlToPdfDocument()
+            {
+                GlobalSettings = globalSettings,
+                Objects = { objectSettings }
+            };
+
+
+            var result = converter.Convert(pdf);
+            return result;
+        }
+        public static byte[] WriteCSVFile(List<Transaction> transactions, string name, string period)
+        {
+            var data = new List<ExportTransactionResponse>();
+            foreach (var item in transactions)
+            {
+
+                data.Add(
+                   new ExportTransactionResponse
+                   {
+                       Date = item.TransactionDate.ToString("dd MMM yyyy"),
+                       TransactionDetail = item.TransactionSource,
+                       Debit = item.PaidAmount.ToString(),
+                       Credit = item.HousingBenefitAmount.ToString(),
+                       Balance = item.BalanceAmount.ToString()
+                   });
+            }
+
+            byte[] result;
+            var cc = new CsvConfiguration(new System.Globalization.CultureInfo("en-US"));
+            using (var ms = new MemoryStream())
+            {
+                using (var sw = new StreamWriter(stream: ms, encoding: new UTF8Encoding(true)))
+                {
+                    using var cw = new CsvWriter(sw, cc);
+                    cw.WriteRecords(data);
+                    cw.WriteComment($"{name} STATEMENT OF YOUR ACCOUNT");
+                    cw.WriteComment($"for the period {period}");
+                    cw.WriteComment($"As of {DateTime.Today:D} your account balance was {transactions.LastOrDefault().BalanceAmount} in arrears.");
+                    cw.WriteComment("As your landlord, the council has a duty to make sure all charges are paid up to date. This is because the housing income goes toward the upkeep of council housing and providing services for residents. You must make weekly charges payment a priority. If you don’t pay, you risk losing your home.");
```

Comment:
Can those writings not be hardcoded to make it more versatile (for different use cases) as well as easy to change without having to redeploy the whole API?